### ISSUE TITLE
IDISP013 What's the motivation for it?

### ISSUE THREAD

**Issue by Bouke** [DotNetAnalyzers/IDisposableAnalyzers#233]
Some code is triggering IDISP013, but no further information is available on why this is a problem. It would be great if some explanation could be added to https://github.com/DotNetAnalyzers/IDisposableAnalyzers/blob/master/documentation/IDISP013.md.

Is the following a good suggestion to resolve this issue?
```c#
// bad
public Task<string> Foo()
{
    using(var bar = new Bar())
        return bar.DoAsync();
}

// good
public async Task<string> Foo()
{
    using(var bar = new Bar())
        return await bar.DoAsync();
}
```

**Comment by JohanLarsson** [comment]
I'll add some documentation.
Your example is correct. The problem is that `bar` is disposed when returning from the method and that is likely to be before the task is completed which can lead to `ObjectDisposedException`depending on implementation.
